network/test/middleware_test.go

```diff
@@ -284,11 +284,28 @@ func (m *MiddlewareTestSuite) TestLargeMessageSize_SendDirect() {
 	encodedEvent, err := codec.Encode(event)
 	require.NoError(m.T(), err)
 
+	// set the message payload as the large message
 	msg.Payload = encodedEvent
 
-	// sends a direct message from first node to the last node
-	err = m.mws[first].SendDirect(msg, lastNode)
+	// expect one message to be received by the target
+	ch := make(chan struct{})
+	m.ov[targetIndex].On("Receive", sourceNode, msg).Return(nil).Once().
+		Run(func(args mockery.Arguments) {
+			ch <- struct{}{}
+		})
+
+	// sends a direct message from source node to the target node
+	err = m.mws[sourceIndex].SendDirect(msg, targetNode)
+	// SendDirect should not error since this is a known large message
 	require.NoError(m.Suite.T(), err)
```

Comment:
In this test, the message made it through to the target node and the target middleware attempted to call `overlay.receive` to delivery the message to the overlay. However, I hadn't defined any mock for `overlay.receive` and every once in while the test failed will unexpected call error. 
Now added an explicit `overlay.receive` mock and assert to check the message was indeed received.